extensions/sockets/src/main/java/org/mule/module/socket/api/client/TcpRequesterClient.java

```diff
@@ -0,0 +1,135 @@
+import org.mule.module.socket.api.protocol.TcpProtocol;
+import org.mule.module.socket.api.tcp.TcpClientSocketProperties;
+import org.mule.runtime.api.connection.ConnectionException;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketException;
+
+public class TcpRequesterClient extends AbstractSocketClient implements RequesterSocket
+{
+
+    private Socket socket;
+    private TcpClientSocketProperties socketProperties;
+    private TcpProtocol protocol;
+
+    public TcpRequesterClient(TcpClientSocketProperties socketProperties, TcpProtocol protocol, String host, Integer port) throws ConnectionException
+    {
+        super(host, port);
+        this.socket = new Socket();
+        this.protocol = protocol;
+        this.socketProperties = socketProperties;
+        configureSocket();
+    }
+
+    private void configureSocket() throws ConnectionException
+    {
+        try
+        {
+            if (socketProperties.getKeepAlive() != null)
+            {
+                socket.setKeepAlive(socketProperties.getKeepAlive());
+            }
+
+            if (socketProperties.getSendBufferSize() != null)
+            {
+                socket.setSendBufferSize(socketProperties.getSendBufferSize());
+            }
+
+            if (socketProperties.getReceiveBufferSize() != null)
+            {
+                socket.setReceiveBufferSize(socketProperties.getReceiveBufferSize());
+            }
+
+            if (socketProperties.getLinger() != null)
+            {
+                socket.setSoLinger(true, socketProperties.getLinger());
+            }
+
+
+            socket.setTcpNoDelay(socketProperties.getSendTcpNoDelay());
+            socket.setSoTimeout(socketProperties.getTimeout());
+            socket.bind(null);
+        }
+        catch (SocketException e)
+        {
+            e.printStackTrace();
+            throw new ConnectionException("Could not create socket due to a SocketException");
+        }
+
+        catch (IOException e)
+        {
+            e.printStackTrace();
+            throw new ConnectionException("Could not create socket due to a IOException");
+        }
+
+    }
+
+    public void connect() throws ConnectionException, UnresolvableHostException
+    {
+
+        try
+        {
+            InetSocketAddress address = getSocketAddress(socketProperties.getFailOnUnresolvedHost(), host, port);
+            // TODO validate address InetAddressValidator from Apache commons
+            socket.connect(address);
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+            throw new ConnectionException(String.format("Could not connect to host '%s' on port '%d'", host, port));
+        }
+    }
+
+    public void write(Object data) throws ConnectionException
+    {
+        BufferedOutputStream bos = null;
```

Comment:
use a try..resource block to make sure that bos is always closed. Also rename to outputStream